### PR TITLE
Add more documentation about a case where .offer() can throw

### PR DESCRIPTION
**Pull request by martinbonnin**
Calling offer on a closed channel will throw. See https://github.com/Kotlin/kotlinx.coroutines/issues/974.

Make this explicit in the code snippets.

### CODE REVIEW

File: kotlinx-coroutines-core/common/src/flow/Builders.kt [hunk 1]
```diff
@@ -293,9 +293,14 @@ public fun <T> channelFlow(@BuilderInference block: suspend ProducerScope<T>.()
...
  *     val callback = object : Callback { // implementation of some callback interface
  *         override fun onNextValue(value: T) {
- *             // Note: offer drops value when buffer is full
+ *             // Note: offer drops values when the buffer is full
  *             // Use either buffer(Channel.CONFLATED) or buffer(Channel.UNLIMITED) to avoid overfill
- *             offer(value)
+ *             // Also, offer will throw if the channel is canceled.
+ *             // To avoid that, we wrap the call in runBlocking.
+ *             // See https://github.com/Kotlin/kotlinx.coroutines/issues/974
+ *             runBlocking {
```
Comment (LouisCAD) [inline kotlinx-coroutines-core/common/src/flow/Builders.kt]:
```suggestion
 *             runCatching {
```

Comment (martinbonnin) [reply kotlinx-coroutines-core/common/src/flow/Builders.kt]: Oh my :see_no_evil:... Thanks :pray:  !

**Comment by elizarov** [comment]
This PR is now in conflict since the corresponding callback flow example had been update.

**Comment by LouisCAD** [comment]
Also, maybe it makes sense to change the target branch to `develop`?

**Comment by martinbonnin** [comment]
It looks like `develop` has the `try/catch` already: https://github.com/Kotlin/kotlinx.coroutines/blob/develop/kotlinx-coroutines-core/common/src/flow/Builders.kt#L293

(from https://github.com/Kotlin/kotlinx.coroutines/commit/4aa38803af953684e4f670d7f30d78d277c486f0#diff-57fe61dee82c96bd815e588a3af51853R286)

I'm closing this PR

**Pull request closed without merging**